logical/src/main/java/org/apache/drill/common/expression/ExpressionStringBuilder.java

```diff
@@ -119,14 +126,14 @@ public Void visitSchemaPath(SchemaPath path, StringBuilder sb) throws RuntimeExc
       throw new IllegalStateException("Drill doesn't currently support top level arrays");
     }
     sb.append('`');
-    sb.append(seg.getNameSegment().getPath());
+    sb.append(escapeBackTick(seg.getNameSegment().getPath()));
```

Comment:
Should getPath() ever return something that cannot be parsed, should we just add this behavior there?